shcomplete/seq2seq_prediction.py

```diff
@@ -0,0 +1,198 @@
+    while True:
+        with open(path_to_corpus, "r") as f:
+            histories = f.read().split(file_delimiter)
+            for history in histories:
+                history = history.split("\n")
+                for _ in range(gConfig["batch_size"]):
+                    ind = random.randint(0, len((history[:-gConfig["seq_len"]])) - 1)
+                    sequence = [cmd.rstrip() for cmd in history[ind:ind + gConfig["seq_len"]]]
+                    sequences.append(sequence)
+                    vocab_trie = vocab.trie(path_to_vocab)
+                    next_prefix = vocab_trie.longest_prefix(history[ind + gConfig["seq_len"]])[0]
+                    if next_prefix is not None and next_prefix:
+                        next_commands.append(next_prefix)
+                    else:
+                        next_commands.append("UNK")
+
+                    if len(sequences) == gConfig["batch_size"]:
+                        X, y = vectorize(sequences, next_commands, vocab, path_to_vocab)
+                        yield X, y
+                        sequences = []
+                        next_commands = []
+
+
+def generate_model(vocab):
+    """
+    Generate the model.
+    """
+    model = Sequential()
+    for layer_id in range(gConfig["input_layers"]):
+        model.add(LSTM(gConfig["hidden_size"], input_shape=(gConfig["seq_len"], vocab.size),
+                       return_sequences=layer_id + 1 < gConfig["input_layers"]))
+        model.add(Dropout(gConfig["amount_of_dropout"]))
+
+    model.add(Dense(vocab.size))
+    model.add(Activation("softmax"))
+    optimizer = RMSprop(lr=gConfig["learning_rate"])
+    model.compile(loss="categorical_crossentropy", optimizer="adam")
+    return model
+
+
+def sample_prediction(model, X, y, vocab):
+    """
+    Select a sequence of shell commands and print the prediction of the model at current training.
+    """
+    index = random.randint(0, len(X)-1)
+    rowX = X[np.array([index])]
+    rowy = y[np.array([index])]
+
+    sequence = vocab.decode(rowX[0])
+    preds = model.predict(rowX, verbose=0)[0]
+    preds = np.reshape(preds, (1, vocab.size))
+    y = np.reshape(rowy[0], (1, vocab.size))
+    next_command_pred = vocab.decode(preds)
+    next_command_true = vocab.decode(y)
+
+    print("----- Generating from seed -----")
+    sys.stdout.write(sequence + "\n")
+
+    print("----- Next command prediction -----")
+    sys.stdout.write(next_command_pred + "\n")
+
+    print("----- Next command True -----")
+    sys.stdout.write(next_command_true + "\n")
+    sys.stdout.flush()
+    print()
+
+
+class OnEpochEndCallback(Callback):
+
+    def on_epoch_end(self, epoch, logs=None):
+        """
+        Apply the prediction of the model to a batch of data at the end of each epoch.
+        """
+        vocab = vocabulary("vocab_0.01.txt")
+        X_batch, y_batch = next(generator("corpus3.txt", "FILE_SEP\n", "vocab_0.01.txt"))
+        sample_prediction(self.model, X_batch, y_batch, vocab)
```

Comment:
I have an issue with this function. It does not accept additional arguments whereas all those magic values are in the args